src/dev-utils/view.js

```diff
@@ -453,7 +457,7 @@ class RangeParser {
 				// Non-empty text node.
 				if ( text ) {
 					if (
-						( this.sameSelectionCharacters && !item.outer ) ||
+						this.sameSelectionCharacters ||
```

Comment:
This condition was not needed and was not covered by tests. I've run tests for entire CKE5 after removing it and everything was green.